### PR TITLE
[NUI] Fix CA1819 : Properties should not return arrays

### PR DESCRIPTION
**Pull request by AdunFang**
### Description of Change ###

Fix CA1819.
https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1819

### API Changes ###

 - ACR: https://code.sec.samsung.net/jira/browse/TCSACR-395

### CODE REVIEW

**Review by zhouleonlei — APPROVED** [review]
LGTM

**Review by AchoWang — APPROVED** [review]
Looks good to me.

**Review by huayongxu — APPROVED** [review]
looks good to me.

**Review by elishateng — APPROVED** [review]
LGTM

File: src/Tizen.NUI/src/public/Animation.cs [hunk 1]
```diff
@@ -434,6 +440,8 @@ public float ProgressNotification
         /// <summary>
         /// Gets or sets the properties of the animation.
         /// </summary>
+        [Obsolete("Deprecated in API6, Will be removed in API9, Please use as keyword instead!")]
+        [SuppressMessage("Microsoft.Naming", "CA1819:This API will be deprecated, so suppressing the warning for now")]
```
Comment (huiyueun) [inline src/Tizen.NUI/src/public/Animation.cs:444]: deprecated ACR is required and you have to write a comment based on the current version.

File: src/Tizen.NUI/src/public/Animation.cs [hunk 2]
```diff
@@ -55,6 +56,11 @@ public class Animation : BaseHandle
         private int[] _startTime = null;
         private int[] _endTime = null;
 
+        private List<string> _propertiesList = null;
```
Comment (hyunjushin) [inline src/Tizen.NUI/src/public/Animation.cs:59]: _propertyList is correct

File: src/Tizen.NUI/src/public/Animation.cs [hunk 3]
```diff
@@ -491,6 +505,78 @@ public int[] EndTime
             }
         }
 
+        /// <summary>
+        /// Get the list of the properties of the animation.
+        /// </summary>
+        /// This will be public opened in tizen_next after ACR done. Before ACR, need to be hidden as inhouse API.
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public List<string> PropertiesList
```
Comment (hyunjushin) [inline src/Tizen.NUI/src/public/Animation.cs:513]: PropertyList is correct

File: src/Tizen.NUI/src/public/Animation.cs [hunk 4]
```diff
@@ -434,6 +440,8 @@ public float ProgressNotification
         /// <summary>
         /// Gets or sets the properties of the animation.
         /// </summary>
+        [Obsolete("Deprecated in API6, Will be removed in API9, Please use as keyword instead!")]
```
Comment (hyunjushin) [inline src/Tizen.NUI/src/public/Animation.cs:443]: Obsolete("Deprecated in API9, Will be removed in API11, Please use PropertyList instead")

**Review by hyunjushin — CHANGES_REQUESTED** [review]

File: src/Tizen.NUI/src/public/Animation.cs (hunk 4 above)
Comment (hyunjushin) [reply src/Tizen.NUI/src/public/Animation.cs:443]: please update every Obsolete information

File: src/Tizen.NUI/src/public/BaseComponents/View.cs [hunk 5]
```diff
@@ -2459,5 +2444,13 @@ public virtual void ApplyStyle(ViewStyle viewStyle)
                 }
             }
         }
+
+        /// This will be public opened in tizen_next after ACR done. Before ACR, need to be hidden as inhouse API.
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public StoryBoard StoryBoard
```
Comment (rabbitfor) [inline src/Tizen.NUI/src/public/BaseComponents/View.cs:2450]: I think this is not related with this patch.

File: src/Tizen.NUI/src/public/XamlBinding/Transition.cs [hunk 6]
```diff
@@ -10,6 +11,18 @@
 
 namespace Tizen.NUI
 {
+    /// This will be public opened in tizen_next after ACR done. Before ACR, need to be hidden as inhouse API.
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class StoryBoard
```
Comment (rabbitfor) [inline src/Tizen.NUI/src/public/XamlBinding/Transition.cs:16]: I think this class is not related with this patch.

**Review by rabbitfor — COMMENTED** [review]
Please check 2 build warnings in Transition.cs:119

**Comment by rabbitfor** [comment]
@AdunFang 
Please check out 2 build warnings appear in Transition.cs:119

**Comment by dongsug-song** [comment]
@AdunFang 
Hello,
please do not upload ACR-PR in DevelNUI branch.
here is a tip to handle ACR-PR here, I hope this helps.
thank you.

- Add-ACR : 
  1) the ACR-PR should be processed in master branch only.
  2) so, make API as hidden by adding `[EditorBrowsable(EditorBrowsableState.Never)]` attribute, then in DevelNUI branch API-Checker will pass. (No "ACR Required" tag)
  3) after the patch is merged into master branch by regular integration work, you can make new PR in master branch by removing `[EditorBrowsable(EditorBrowsableState.Never)]` attribute to proceed ACR.
  4) if the ACR is urgent, you can upload PR on master branch directly. that PR will be merged on DevelNUI branch by rebasing during the regular integration work.

- Deprecate-ACR : 
  1) the ACR-PR should be processed in master branch only.
  2) so, add some mark on the API by writing some comments like `//ToDo : will raise deprecated-ACR, [Obsolete("This has been deprecated in APIxxx and will be removed in APIxxx+2. The xxx variable is not used anymore and the xxx is replaced by xxx! Please use xxx instead!")`
  3) by step 2), the PR on DevelNUI will pass. (No "ACR Required" tag)
  4) after the patch is merged into master branch by regular integration work, you can make new PR in master branch by adding `[Obsolete("blah blah")]` tag in order to proceed ACR.
  5) if the ACR is urgent, you can upload PR on master branch directly. that PR will be merged on DevelNUI branch by rebasing during the regular integration work.

File: src/Tizen.NUI/src/public/BaseComponents/View.cs (hunk 5 above)
Comment (AdunFang) [reply src/Tizen.NUI/src/public/BaseComponents/View.cs:2450]: To fix the CA1819, I remove the property TranstionNames,
and because the transition has been refactor, so I use a new class to restore the transtion.

The name "StoryBoard" is refer to WPF.

File: src/Tizen.NUI/src/public/XamlBinding/Transition.cs (hunk 6 above)
Comment (AdunFang) [reply src/Tizen.NUI/src/public/XamlBinding/Transition.cs:16]: To fix the CA1819, I remove the property TranstionNames,
and because the transition has been refactor, so I use a new class to restore the transtion.

The name "StoryBoard" is refer to WPF.

File: src/Tizen.NUI/src/public/Animation.cs [hunk 7]
```diff
@@ -436,6 +442,8 @@ public float ProgressNotification
         /// <summary>
         /// Gets or sets the properties of the animation.
         /// </summary>
+        [Obsolete("Deprecated in API9, Will be removed in API11, Please use PropertyList instead")]
```
Comment (dongsug-song) [inline src/Tizen.NUI/src/public/Animation.cs:445]: Hello, 
I can see this is for Deprecated-ACR. 
please follow the steps below.
1) add comments like : `//ToDo : will raise deprecated-ACR, [Obsolete("Deprecated in API9, Will be removed in API11. Please use PropertyList instead!")]`
2) get review in DevelNUI with **No "ACR Required" tag**
3) when this is merged in master branch by integration work, you can modify the comment of the `[Obsolete(...)]` tag to **be activated** and use it for ACR process (need to submit new PR in master branch)
  (example) `//ToDo : will raise deprecated-ACR, [Obsolete("...")]` => `[Obsolete("...")]`
4) the PR for ACR should be reviewed on **master branch only**
thank you.

File: src/Tizen.NUI/src/public/Animation.cs [hunk 8]
```diff
@@ -451,6 +459,8 @@ public string[] Properties
         /// <summary>
         /// Gets or sets the destination value for each property of the animation.
         /// </summary>
+        [Obsolete("Deprecated in API9, Will be removed in API11, Please use DestValueList instead")]
```
Comment (dongsug-song) [inline src/Tizen.NUI/src/public/Animation.cs:462]: Hello, 
I can see this is for Deprecated-ACR. 
please follow the steps below.
1) add comments like : `//ToDo : will raise deprecated-ACR, [Obsolete("Deprecated in API9, Will be removed in API11. Please use DestValueList instead!")]`
2) get review in DevelNUI with **No "ACR Required" tag**
3) when this is merged in master branch by integration work, you can modify the comment of the `[Obsolete(...)]` tag to **be activated** and use it for ACR process (need to submit new PR in master branch)
  (example) `//ToDo : will raise deprecated-ACR, [Obsolete("...")]` => `[Obsolete("...")]`
4) the PR for ACR should be reviewed on **master branch only**
thank you.

File: src/Tizen.NUI/src/public/Animation.cs [hunk 9]
```diff
@@ -466,6 +476,8 @@ public string[] DestValue
         /// <summary>
         /// Gets or sets the start time for each property of the animation.
         /// </summary>
+        [Obsolete("Deprecated in API9, Will be removed in API11, Please use StartTimeList instead")]
```
Comment (dongsug-song) [inline src/Tizen.NUI/src/public/Animation.cs:479]: Hello, 
I can see this is for Deprecated-ACR. 
please follow the steps below.
1) add comments like : `//ToDo : will raise deprecated-ACR, [Obsolete("Deprecated in API9, Will be removed in API11. Please use StartTimeList instead!")]`
2) get review in DevelNUI with **No "ACR Required" tag**
3) when this is merged in master branch by integration work, you can modify the comment of the `[Obsolete(...)]` tag to **be activated** and use it for ACR process (need to submit new PR in master branch)
  (example) `//ToDo : will raise deprecated-ACR, [Obsolete("...")]` => `[Obsolete("...")]`
4) the PR for ACR should be reviewed on **master branch only**
thank you.

File: src/Tizen.NUI/src/public/Animation.cs [hunk 10]
```diff
@@ -481,6 +493,8 @@ public int[] StartTime
         /// <summary>
         /// Gets or sets the end time for each property of the animation.
         /// </summary>
+        [Obsolete("Deprecated in API9, Will be removed in API11, Please use EndTimeList instead")]
```
Comment (dongsug-song) [inline src/Tizen.NUI/src/public/Animation.cs:496]: Hello, 
I can see this is for Deprecated-ACR. 
please follow the steps below.
1) add comments like : `//ToDo : will raise deprecated-ACR, [Obsolete("Deprecated in API9, Will be removed in API11. Please use EndTimeList instead!")]`
2) get review in DevelNUI with **No "ACR Required" tag**
3) when this is merged in master branch by integration work, you can modify the comment of the `[Obsolete(...)]` tag to **be activated** and use it for ACR process (need to submit new PR in master branch)
  (example) `//ToDo : will raise deprecated-ACR, [Obsolete("...")]` => `[Obsolete("...")]`
4) the PR for ACR should be reviewed on **master branch only**
thank you.

File: src/Tizen.NUI/src/public/Animation.cs [hunk 11]
```diff
@@ -493,6 +507,78 @@ public int[] EndTime
             }
         }
 
+        /// <summary>
+        /// Get the list of the properties of the animation.
+        /// </summary>
+        /// This will be public opened in tizen_next after ACR done. Before ACR, need to be hidden as inhouse API.
```
Comment (dongsug-song) [inline src/Tizen.NUI/src/public/Animation.cs:513]: In TCSACR-395, I couldn't see any "Add-ACR". so I understand that this is added as Hidden-API, then please **REMOVE** this comment. : `/// This will be public opened in tizen_next after ACR done. Before ACR, need to be hidden as inhouse API.`
thank you.

File: src/Tizen.NUI/src/public/Animation.cs [hunk 12]
```diff
@@ -493,6 +507,78 @@ public int[] EndTime
...
+                }
+
+                return _propertyList;
+            }
+        }
+
+        /// <summary>
+        /// Get the list of the destination value for each property of the animation.
+        /// </summary>
+        /// This will be public opened in tizen_next after ACR done. Before ACR, need to be hidden as inhouse API.
```
Comment (dongsug-song) [inline src/Tizen.NUI/src/public/Animation.cs:531]: In TCSACR-395, I couldn't see any "Add-ACR". so I understand that this is added as Hidden-API, then please **REMOVE** this comment. : `/// This will be public opened in tizen_next after ACR done. Before ACR, need to be hidden as inhouse API.`
thank you.

File: src/Tizen.NUI/src/public/Animation.cs [hunk 13]
```diff
@@ -493,6 +507,78 @@ public int[] EndTime
...
+                }
+
+                return _destValueList;
+            }
+        }
+
+        /// <summary>
+        /// Get the list of the start time for each property of the animation.
+        /// </summary>
+        /// This will be public opened in tizen_next after ACR done. Before ACR, need to be hidden as inhouse API.
```
Comment (dongsug-song) [inline src/Tizen.NUI/src/public/Animation.cs:549]: In TCSACR-395, I couldn't see any "Add-ACR". so I understand that this is added as Hidden-API, then please **REMOVE** this comment. : `/// This will be public opened in tizen_next after ACR done. Before ACR, need to be hidden as inhouse API.`
thank you.

File: src/Tizen.NUI/src/public/Animation.cs [hunk 14]
```diff
@@ -493,6 +507,78 @@ public int[] EndTime
...
+                }
+
+                return _startTimeList;
+            }
+        }
+
+        /// <summary>
+        /// Get the list of end time for each property of the animation.
+        /// </summary>
+        /// This will be public opened in tizen_next after ACR done. Before ACR, need to be hidden as inhouse API.
```
Comment (dongsug-song) [inline src/Tizen.NUI/src/public/Animation.cs:567]: In TCSACR-395, I couldn't see any "Add-ACR". so I understand that this is added as Hidden-API, then please **REMOVE** this comment. : `/// This will be public opened in tizen_next after ACR done. Before ACR, need to be hidden as inhouse API.`
thank you.

File: src/Tizen.NUI/src/public/BaseComponents/View.cs [hunk 15]
```diff
@@ -2487,5 +2471,15 @@ public virtual void ApplyStyle(ViewStyle viewStyle)
                 }
             }
         }
+
+        /// <summary>
+        /// Used to restore the transition.
+        /// </summary>
+        /// This will be public opened in tizen_next after ACR done. Before ACR, need to be hidden as inhouse API.
```
Comment (dongsug-song) [inline src/Tizen.NUI/src/public/BaseComponents/View.cs:2478]: In TCSACR-395, I couldn't see any "Add-ACR". so I understand that this is added as Hidden-API, then please **REMOVE** this comment. : `/// This will be public opened in tizen_next after ACR done. Before ACR, need to be hidden as inhouse API.`
thank you.

File: src/Tizen.NUI/src/public/XamlBinding/Transition.cs [hunk 16]
```diff
@@ -11,7 +12,7 @@
 namespace Tizen.NUI
 {
     /// <since_tizen> 5 </since_tizen>
-    /// This will be public opened in tizen_5.0 after ACR done. Before ACR, need to be hidden as inhouse API.
+    /// This will be public opened in tizen_next after ACR done. Before ACR, need to be hidden as inhouse API.
```
Comment (dongsug-song) [inline src/Tizen.NUI/src/public/XamlBinding/Transition.cs:15]: In TCSACR-395, I couldn't see any "Add-ACR". so I understand that this is added as Hidden-API, then please **REMOVE** this comment. : `/// This will be public opened in tizen_next after ACR done. Before ACR, need to be hidden as inhouse API.`
thank you.

File: src/Tizen.NUI/src/public/XamlBinding/Transition.cs [hunk 17]
```diff
@@ -100,6 +101,45 @@ public int EndTime
         }
     }
 
+
+    /// <summary>
+    /// It is the container to contain the behaviors of Transition.
+    /// </summary>
+    /// This will be public opened in tizen_6.5 after ACR done. Before ACR, need to be hidden as inhouse API.
```
Comment (dongsug-song) [inline src/Tizen.NUI/src/public/XamlBinding/Transition.cs:108]: In TCSACR-395, I couldn't see any "Add-ACR". so I understand that this is added as Hidden-API, then please **REMOVE** this comment. : `/// This will be public opened in tizen_next after ACR done. Before ACR, need to be hidden as inhouse API.`
thank you.

File: src/Tizen.NUI/src/public/XamlBinding/Transition.cs [hunk 18]
```diff
@@ -100,6 +101,45 @@ public int EndTime
...
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class BehaviorContainer : List<AnimationBehavior>
+    {
+        private Dictionary<string, AnimationBehavior> behaviors = new Dictionary<string, AnimationBehavior>();
+
+        /// <summary>
+        /// The method for user to add behavior.
+        /// </summary>
+        /// This will be public opened in tizen_6.5 after ACR done. Before ACR, need to be hidden as inhouse API.
```
Comment (dongsug-song) [inline src/Tizen.NUI/src/public/XamlBinding/Transition.cs:117]: In TCSACR-395, I couldn't see any "Add-ACR". so I understand that this is added as Hidden-API, then please **REMOVE** this comment. : `/// This will be public opened in tizen_next after ACR done. Before ACR, need to be hidden as inhouse API.`
thank you.

File: src/Tizen.NUI/src/public/XamlBinding/Transition.cs [hunk 19]
```diff
@@ -100,6 +101,45 @@ public int EndTime
...
+            if (null != behavior)
+            {
+                behaviors.Add(behavior.Key, behavior);
+            }
+        }
+
+        /// <summary>
+        /// The method for user to get the behavior by the key.
+        /// </summary>
+        /// This will be public opened in tizen_6.5 after ACR done. Before ACR, need to be hidden as inhouse API.
```
Comment (dongsug-song) [inline src/Tizen.NUI/src/public/XamlBinding/Transition.cs:132]: In TCSACR-395, I couldn't see any "Add-ACR". so I understand that this is added as Hidden-API, then please **REMOVE** this comment. : `/// This will be public opened in tizen_next after ACR done. Before ACR, need to be hidden as inhouse API.`
thank you.

File: src/Tizen.NUI/src/public/XamlBinding/Transition.cs [hunk 20]
```diff
@@ -212,5 +231,13 @@ public void AnimateBy(View instance, string behaviorKey)
                 throw new XamlParseException(string.Format("Behaviors don't have key {0}", behaviorKey), new XmlLineInfo());
             }
         }
+
+        /// <since_tizen> 5 </since_tizen>
+        /// This will be public opened in tizen_6.5 after ACR done. Before ACR, need to be hidden as inhouse API.
```
Comment (dongsug-song) [inline src/Tizen.NUI/src/public/XamlBinding/Transition.cs:236]: In TCSACR-395, I couldn't see any "Add-ACR". so I understand that this is added as Hidden-API, then please **REMOVE** this comment. : `/// This will be public opened in tizen_next after ACR done. Before ACR, need to be hidden as inhouse API.`
thank you.

File: src/Tizen.NUI/src/public/Animation.cs [hunk 21]
```diff
@@ -493,6 +507,78 @@ public int[] EndTime
...
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public List<string> PropertyList
+        {
+            get
+            {
+                if (null == _propertyList)
+                {
+                    _propertyList = new List<string>();
```
Comment (dongsug-song) [inline src/Tizen.NUI/src/public/Animation.cs:521]: in c#, pascal case is supposed to be used and for private member, it would be OK to use `_` prefix. However in NUI, we have decided **NOT to use prefix such as `_`, `m`, `p`, and etc.**

- NUI ground rule : https://code.sec.samsung.net/confluence/display/GFX/NUI+Ground+Rule
- NUI coding rule : https://code.sec.samsung.net/confluence/pages/viewpage.action?pageId=191373542

could you use **PascalCase** even for private fields?

File: src/Tizen.NUI/src/public/Animation.cs [hunk 22]
```diff
@@ -493,6 +507,78 @@ public int[] EndTime
...
+        public List<int> EndTimeList
+        {
+            get
+            {
+                if (null == _endTimeList)
+                {
+                    _endTimeList = new List<int>();
+                }
+
+                return _endTimeList;
```
Comment (dongsug-song) [inline src/Tizen.NUI/src/public/Animation.cs:578]: in c#, pascal case is supposed to be used and for private member, it would be OK to use `_` prefix. However in NUI, we have decided **NOT to use prefix such as `_`, `m`, `p`, and etc.**

- NUI ground rule : https://code.sec.samsung.net/confluence/display/GFX/NUI+Ground+Rule
- NUI coding rule : https://code.sec.samsung.net/confluence/pages/viewpage.action?pageId=191373542

could you use **PascalCase** even for private fields?

File: src/Tizen.NUI/src/public/Animation.cs [hunk 23]
```diff
@@ -493,6 +507,78 @@ public int[] EndTime
             }
         }
 
+        /// <summary>
+        /// Get the list of the properties of the animation.
+        /// </summary>
+        /// This will be public opened in tizen_next after ACR done. Before ACR, need to be hidden as inhouse API.
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public List<string> PropertyList
```
Comment (hyunjushin) [inline src/Tizen.NUI/src/public/Animation.cs:515]: Framework Design Guidelines don't recommend using List<T> in public APIs.

https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/guidelines-for-collections

❌ DO NOT use ArrayList or List in public APIs.
These types are data structures designed to be used in internal implementation, not in public APIs. List is optimized for performance and power at the cost of cleanness of the APIs and flexibility. For example, if you return List, you will not ever be able to receive notifications when client code modifies the collection. Also, List exposes many members, such as BinarySearch, that are not useful or applicable in many scenarios. The following two sections describe types (abstractions) designed specifically for use in public APIs.

I think you can use IList<T> instead of List<T>.

File: src/Tizen.NUI/src/public/Animation.cs [hunk 24]
```diff
@@ -493,6 +507,78 @@ public int[] EndTime
...
+                return _propertyList;
+            }
+        }
+
+        /// <summary>
+        /// Get the list of the destination value for each property of the animation.
+        /// </summary>
+        /// This will be public opened in tizen_next after ACR done. Before ACR, need to be hidden as inhouse API.
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public List<string> DestValueList
```
Comment (hyunjushin) [inline src/Tizen.NUI/src/public/Animation.cs:533]: Framework Design Guidelines don't recommend using List<T> in public APIs.

https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/guidelines-for-collections

❌ DO NOT use ArrayList or List in public APIs.
These types are data structures designed to be used in internal implementation, not in public APIs. List is optimized for performance and power at the cost of cleanness of the APIs and flexibility. For example, if you return List, you will not ever be able to receive notifications when client code modifies the collection. Also, List exposes many members, such as BinarySearch, that are not useful or applicable in many scenarios. The following two sections describe types (abstractions) designed specifically for use in public APIs.

I think you can use IList<T> instead of List<T>.

File: src/Tizen.NUI/src/public/Animation.cs [hunk 25]
```diff
@@ -493,6 +507,78 @@ public int[] EndTime
...
+                return _destValueList;
+            }
+        }
+
+        /// <summary>
+        /// Get the list of the start time for each property of the animation.
+        /// </summary>
+        /// This will be public opened in tizen_next after ACR done. Before ACR, need to be hidden as inhouse API.
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public List<int> StartTimeList
```
Comment (hyunjushin) [inline src/Tizen.NUI/src/public/Animation.cs:551]: Framework Design Guidelines don't recommend using List<T> in public APIs.

https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/guidelines-for-collections

❌ DO NOT use ArrayList or List in public APIs.
These types are data structures designed to be used in internal implementation, not in public APIs. List is optimized for performance and power at the cost of cleanness of the APIs and flexibility. For example, if you return List, you will not ever be able to receive notifications when client code modifies the collection. Also, List exposes many members, such as BinarySearch, that are not useful or applicable in many scenarios. The following two sections describe types (abstractions) designed specifically for use in public APIs.

I think you can use IList<T> instead of List<T>.

File: src/Tizen.NUI/src/public/Animation.cs [hunk 26]
```diff
@@ -493,6 +507,78 @@ public int[] EndTime
...
+                return _startTimeList;
+            }
+        }
+
+        /// <summary>
+        /// Get the list of end time for each property of the animation.
+        /// </summary>
+        /// This will be public opened in tizen_next after ACR done. Before ACR, need to be hidden as inhouse API.
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public List<int> EndTimeList
```
Comment (hyunjushin) [inline src/Tizen.NUI/src/public/Animation.cs:569]: Framework Design Guidelines don't recommend using List<T> in public APIs.

https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/guidelines-for-collections

❌ DO NOT use ArrayList or List in public APIs.
These types are data structures designed to be used in internal implementation, not in public APIs. List is optimized for performance and power at the cost of cleanness of the APIs and flexibility. For example, if you return List, you will not ever be able to receive notifications when client code modifies the collection. Also, List exposes many members, such as BinarySearch, that are not useful or applicable in many scenarios. The following two sections describe types (abstractions) designed specifically for use in public APIs.

I think you can use IList<T> instead of List<T>.

File: src/Tizen.NUI/src/public/BaseComponents/View.cs [hunk 27]
```diff
@@ -2487,5 +2471,15 @@ public virtual void ApplyStyle(ViewStyle viewStyle)
...
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public List<Transition> StoryBoard
```
Comment (hyunjushin) [inline src/Tizen.NUI/src/public/BaseComponents/View.cs:2480]: Framework Design Guidelines don't recommend using List<T> in public APIs.

https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/guidelines-for-collections

❌ DO NOT use ArrayList or List in public APIs.
These types are data structures designed to be used in internal implementation, not in public APIs. List is optimized for performance and power at the cost of cleanness of the APIs and flexibility. For example, if you return List, you will not ever be able to receive notifications when client code modifies the collection. Also, List exposes many members, such as BinarySearch, that are not useful or applicable in many scenarios. The following two sections describe types (abstractions) designed specifically for use in public APIs.

I think you can use IList<T> instead of List<T>.

File: src/Tizen.NUI/src/public/BaseComponents/View.cs [hunk 28]
```diff
@@ -2307,21 +2306,6 @@ public Vector2 UpdateSizeHint
             }
         }
 
-        /// This will be public opened in tizen_next after ACR done. Before ACR, need to be hidden as inhouse API.
```
Comment (hyunjushin) [inline src/Tizen.NUI/src/public/BaseComponents/View.cs:2310]: Can you confirm this property was not used? Hidden api(EditorBrowsableState.Never) are also can be used. Just removing might be dangerous.

File: src/Tizen.NUI/src/public/BaseComponents/View.cs (hunk 27 above)
Comment (hyunjushin) [reply src/Tizen.NUI/src/public/BaseComponents/View.cs:2480]: And not clear the "StoryBoard" name for me.

**Review by hyunjushin — CHANGES_REQUESTED** [review]

**Review by hyunjushin — APPROVED** [review]

File: src/Tizen.NUI/src/public/Animation.cs [hunk 29]
```diff
@@ -436,6 +442,8 @@ public float ProgressNotification
         /// <summary>
         /// Gets or sets the properties of the animation.
         /// </summary>
+        //ToDo : will raise deprecated-ACR, [Obsolete("Deprecated in API9, Will be removed in API11, Please use PropertyList instead")]
+        [SuppressMessage("Microsoft.Naming", "CA1819:This API will be deprecated, so suppressing the warning for now")]
```
Comment (rabbitfor) [inline src/Tizen.NUI/src/public/Animation.cs:446]: Hi @AdunFang ,
The second parameter is `RuleId`, so you may need to add `Justification` to write reason.
https://docs.microsoft.com/en-us/visualstudio/code-quality/in-source-suppression-overview?view=vs-2019#suppressmessage-attribute
```
[SuppressMessage("Rule Category", "Rule Id", Justification = "Justification", MessageId = "MessageId", Scope = "Scope", Target = "Target")]
```

And the rule `CA1819` is not in "Naming" category. It is "Performance".
```
[SuppressMessage("Microsoft.Performance, "CA1819:PropertiesShouldNotReturnArrays", Justification = "This API will be deprecated, so suppressing the warning for now")]
```

**Review by rabbitfor — APPROVED** [review]
LGTM
